dv/formal/Makefile

```diff
@@ -32,33 +40,52 @@ fusesoc:
 	  --tool vcs \
 	  --setup \
 	  lowrisc:ibex:ibex_formal:0.1
+	touch build/fusesoc # Fix timestamps for Makefile
 
-.PHONY: sv
-sv:
+# The sail spec module, constructed from LOWRISC_SAIL_RISCV_SRC, compiled by our fork of the sail compiler
+build/ibexspec.sv: $(SAIL) $(SAIL_SRCS) $(SAIL_EXTRA_SRCS) sail-sources.mk Makefile spec/fix_bugs.py
 	mkdir -p build
-	python3 buildspec.py header > build/ibexspec_instrs.sv
-	cd build && $(SAIL) $(SAIL_SRCS) $(SAIL_EXTRA_SRCS) $(SAIL_SV_FLAGS) `cd .. && python3 buildspec.py unreachables` -o ibexspec
-	python3 spec/fix_pmp_bug.py
-	python3 buildspec.py unreachable_loc_hack
+	cd build && $(SAIL) $(SAIL_SRCS) $(addprefix ../,$(SAIL_EXTRA_SRCS)) $(SAIL_SV_FLAGS) $(addprefix -sv_unreachable execute_,$(COMPRESSED_INSTRS)) -o ibexspec
+	python3 spec/fix_bugs.py
 
-.PHONY: psgen
-psgen:
+# The compiled property structure, and the associated TCL
+build/psgen-jg.sv: $(PSGEN) $(PSGEN_SRCS) Makefile
 	mkdir -p build
-	psgen $(addprefix -path ,$(PSGEN_SRCS)) $(PSGEN_FLAGS) -sv-out build/psgen.sv -tcl-out build/psgen.tcl
+	$(PSGEN) $(addprefix -path ,$(PSGEN_SRCS)) $(PSGEN_FLAGS) -sv-out build/psgen-jg.sv -tcl-out build/psgen.tcl
 
+# Same as above, but generate our own covers (not very well at the minute!), and tweak for consumption by our tools
+build/psgen-yosys.sv: $(PSGEN) $(PSGEN_SRCS) Makefile
+	mkdir -p build
+	$(PSGEN) $(addprefix -path ,$(PSGEN_SRCS)) $(PSGEN_FLAGS) -sv-out build/psgen-yosys.sv -clocking -covers -cover-assert -step-prefix
+
+# The aiger file for the OSS flow, using the yosys_formal/* plugins. The main yosys script is in build_all.ys
+build/all.aig: build/psgen-yosys.sv build/fusesoc build/ibexspec.sv check build_all.ys $(LOWRISC_YOSYS_SLANG) build/yosys_formal.so
+	yosys -m $(LOWRISC_YOSYS_SLANG) -m build/yosys_formal.so -v 1 -s <(cat ./build_all.ys | sed "s|LOWRISC_SAIL_SRC|$(LOWRISC_SAIL_SRC)|g")
```

Comment:
Slightly annoying yosys can't do environment variable substitution, would be nice to not have to do this dance with sed.

Reply:
You might be able to turn `build_all.ys` into a Tcl script and look up the environment via `$::env(FOO)`. Yosys has an embedded Tcl interpreter in most builds